tests/integrations/admin/indexation-integration-test.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+namespace Yoast\WP\SEO\Tests\Integrations\Admin;
+
+use Mockery;
+use Brain\Monkey;
+use WPSEO_Admin_Asset_Manager;
+use Yoast\WP\SEO\Actions\Indexation\Indexable_General_Indexation_Action;
+use Yoast\WP\SEO\Actions\Indexation\Indexable_Post_Indexation_Action;
+use Yoast\WP\SEO\Actions\Indexation\Indexable_Post_Type_Archive_Indexation_Action;
+use Yoast\WP\SEO\Actions\Indexation\Indexable_Term_Indexation_Action;
+use Yoast\WP\SEO\Conditionals\Admin_Conditional;
+use Yoast\WP\SEO\Conditionals\Yoast_Admin_And_Dashboard_Conditional;
+use Yoast\WP\SEO\Helpers\Options_Helper;
+use Yoast\WP\SEO\Tests\TestCase;
+
+use Yoast\WP\SEO\Integrations\Admin\Indexation_Integration;
+
+/**
+ * Class Indexation_Integration_Test
+ *
+ * @group actions
+ * @group indexables
+ *
+ * @coversDefaultClass \Yoast\WP\SEO\Integrations\Admin\Indexation_Integration
+ */
+class Indexation_Integration_Test extends TestCase {
+
+	/**
+	 * Holds indexation integration.
+	 *
+	 * @var Indexation_Integration
+	 */
+	private $instance;
+
+	/**
+	 * Holds the post indexation action mock.
+	 *
+	 * @var Mockery\MockInterface|Indexable_Post_Indexation_Action
+	 */
+	private $post_indexation;
+
+	/**
+	 * Holds the term indexation action mock.
+	 *
+	 * @var Mockery\MockInterface|Indexable_Term_Indexation_Action
+	 */
+	private $term_indexation;
+
+	/**
+	 * Holds the post type archive indexation action mock.
+	 *
+	 * @var Mockery\MockInterface|Indexable_Post_Type_Archive_Indexation_Action
+	 */
+	private $post_type_archive_indexation;
+
+	/**
+	 * Holds the general indexation action mock.
+	 *
+	 * @var Mockery\MockInterface|Indexable_General_Indexation_Action
+	 */
+	private $general_indexation;
+
+	/**
+	 * Holds the options helper mock.
+	 *
+	 * @var Mockery\MockInterface|Options_Helper
+	 */
+	private $options;
+
+	/**
+	 * Holds the admin asset manager mock.
+	 *
+	 * @var Mockery\MockInterface|WPSEO_Admin_Asset_Manager
+	 */
+	private $asset_manager;
+
+
+	/**
+	 * @inheritDoc
+	 */
+	public function setUp() {
+		$this->post_type_archive_indexation = Mockery::mock( Indexable_Post_Type_Archive_Indexation_Action::class );
+		$this->post_indexation              = Mockery::mock( Indexable_Post_Indexation_Action::class );
+		$this->term_indexation              = Mockery::mock( Indexable_Term_Indexation_Action::class );
+		$this->general_indexation           = Mockery::mock( Indexable_General_Indexation_Action::class );
+		$this->options                      = Mockery::mock( Options_Helper::class );
+		$this->asset_manager                = Mockery::mock( WPSEO_Admin_Asset_Manager::class );
+
+		$this->instance = new Indexation_Integration(
+			$this->post_indexation,
+			$this->term_indexation,
+			$this->post_type_archive_indexation,
+			$this->general_indexation,
+			$this->options,
+			$this->asset_manager
+		);
+	}
+
+	/**
+	 * Tests that the indexation is only loaded when:
```

Comment:
In fact this description isn't what your are testing. The test only checks if the method returns an array with two strings.